base/compiler/typelattice.jl

```diff
@@ -118,8 +118,6 @@ function ⊑(@nospecialize(a), @nospecialize(b))
     end
     isa(a, MaybeUndef) && (a = a.typ)
     isa(b, MaybeUndef) && (b = b.typ)
-    (a === NOT_FOUND || b === Any) && return true
```

Comment:
Out of curiosity, did this line previously function as a fast path in the `b === Any` case?

Reply:
(Likewise with `a === Any` below, having determined `b !== Any` on the preceding line?)

Reply:
Ah, I apparently might have deleted too much of this line